### ISSUE TITLE
(WIP) Centralise version number into one place

### ISSUE THREAD

**Issue by damies13** [opentoonz/opentoonz#1368]
In my discussion with #shun-iwasawa while working on PR #1140, we agreed that the version number needs to be centralised in one place to prevent the version number drift we experienced in the tfarm executables.

I am happy to have this issue assigned to me.

**Comment by damies13** [comment]
These are primarily notes to myself, things that this issue needs to address:

1. Consolidate TVER and TEnv
1. update tfarmcontroller to use TEnv
1. update tfarmserver to use TEnv
1. check / update tcomposer to TEnv for version numbers
1. check / update tcleanupper to TEnv for version numbers
1. check / update tconverter to TEnv for version numbers
1. update OpenToonz main GUI application to use TEnv version number information

**Comment by damies13** [comment]
8. update tcli.cpp to use TEnv

**Comment by damies13** [comment]
--- 1. Consolidate TVER and TEnv ---
--- 2. update tfarmcontroller to use TEnv ---
--- 3. update tfarmserver to use TEnv ---
4. check / update tcomposer to TEnv for version numbers
5. check / update tcleanupper to TEnv for version numbers
6. check / update tconverter to TEnv for version numbers
--- 7. update OpenToonz main GUI application to use TEnv version number information ---
--- 8. update tcli.cpp to use TEnv ---

**Comment by konero** [comment]
You can use task lists on git for your notes.

https://help.github.com/articles/about-task-lists/

- [ ] Item 1
- [x] Item 2

_I'll delete this comment later to keep your issue clean._

**Comment by RodneyBaker** [comment]
PR #2410, currently in progress, relates to this.
There are some  lose references to version numbers in various documents and we can list/link those here.

**Comment by RodneyBaker** [comment]
@damies13 Is this still a Work In Progress?
Minor incremental version numbers were retired as of v1.3 so hopefully that resolved the issue(s) here.

**Comment by damies13** [comment]
Honestly, I haven't been able to spend time contributing to this project, but I think the stuff done in v1.3 and that PR you mentioned resolved most of the issues.

I'm not sure but I think some of my PR's around this made it into the code and also helped resolve this. either way I think this was resolved, but I haven't checked.

PR #2410 will have resolved the main reason I raised this, being the toonzfarm components not being included in the OSX version because they ended up in a path outside the main OSX .app file. When I went looking for where I need to make the change to fix this issue I found the version numbers were all over the place.

I'll close this, if someone else thinks it need more work it can be re-opened.
